src/content/app/genome-browser/components/zmenu/Zmenu.tsx

```diff
@@ -54,8 +56,19 @@ export type ZmenuProps = {
 const Zmenu = (props: ZmenuProps) => {
   const anchorRef = useRefWithRerender<HTMLDivElement>(null);
   const { zmenus, setZmenus } = useGenomeBrowser();
+  const actualChrLocation = useAppSelector(getActualChrLocation);
+  const [renderCount, setRenderCount] = useState(0);
+
   const dispatch = useAppDispatch();
 
+  useEffect(() => {
+    setRenderCount(renderCount + 1);
+    // Do not try to close the Zmenu on first render
```

Comment:
omg, this is so ugly! let's see if the `usePrevious` hook helps:

```ts
// no need to call it actual here
const chromosomeLocation = useAppSelector(getActualChrLocation);
const previousChromosomeLocation = usePrevious(chromosomeLocation)

useEffect(() => {
  if (chromosomeLocation.toString() !== previousChromosomeLocation.toString()) {
    destroyZmenu();
  }
}, [chromosomeLocation])
```

Reply:
That was my idea 😃 
`usePrevious` should work as well.